accounts/keys.go

```diff
@@ -0,0 +1,168 @@
+package accounts
+
+import (
+	"crypto/ecdsa"
+	"errors"
+	"io"
+	"os"
+
+	"github.com/howeyc/gopass"
+)
+
+var (
+	errNoKeystoreDir = errors.New("Keystore directory does not exists")
+)
+
+// PassPhraser is interface for retrieving
+// pass phrase for Eth keys
+//
+// If you want to retrieve pass phrases reader different ways
+// (e.g: from file, from env variables, interactively from terminal)
+// you must implement PassPhraser reader a different way and pass it to
+// KeyOpener instance
+type PassPhraser interface {
+	GetPassPhrase() (string, error)
+}
+
+// KeyOpener is interface for loading Eth keys
+type KeyOpener interface {
+	// GetPassPhraser return PassPhraser interface
+	// that provides pass phrase for loaded keys
+	GetPassPhraser() PassPhraser
+	// OpenKeystore opens key storage.
+	OpenKeystore() (bool, error)
+	// GetKey returns private key from opened storage
+	GetKey() (*ecdsa.PrivateKey, error)
+}
+
+// nullPassPhraser implements PassPhraser and always return the same pass phrase.
+// NOTE: Use it for testing purposes only!
+type nullPassPhraser struct{}
+
+func (pf *nullPassPhraser) GetPassPhrase() (string, error) {
+	return "testme", nil
+}
+
+// defaultKeyOpener implements KeyOpener interface
+type defaultKeyOpener struct {
+	keyDirPath string
+	idt        Identity
+	pf         PassPhraser
+}
+
+func (o *defaultKeyOpener) GetPassPhraser() PassPhraser {
+	return o.pf
+}
+
+func (o *defaultKeyOpener) OpenKeystore() (bool, error) {
+	var err error
+	var idt Identity
+
+	defer func() {
+		if err == nil {
+			o.idt = idt
+		}
+	}()
+
+	if !hasDir(o.keyDirPath) {
+		return false, errNoKeystoreDir
+	}
+
+	passPhrase, err := o.pf.GetPassPhrase()
+	if err != nil {
+		return false, err
+	}
+
+	idt = NewIdentity(o.keyDirPath)
+	err = idt.Open(passPhrase)
+	if err == nil {
+		return false, nil
+	}
+
+	if err == ErrWalletIsEmpty {
+		_, err = o.createNewKey(idt)
+		return err == nil, err
```

Comment:
It took me about 5 seconds to figure out what's going on. This is too complicated for typical go developer.